integration/doc/oasis/architecture_design.md

```diff
@@ -0,0 +1,89 @@
+# Architecture design
+
+## Overview
+<img src="https://user-images.githubusercontent.com/16873751/86186667-81bf4480-baee-11ea-9c41-cca538078232.png" alt="overview" width="600"/><br/>
+
+## Service layer
+<img src="https://user-images.githubusercontent.com/16873751/86185744-f1800000-baeb-11ea-8683-4cd8e38a7a4c.png" alt="overview" width="600"/><br/>
+
+## Solution layer
+
+<img src="https://user-images.githubusercontent.com/16873751/86185765-ffce1c00-baeb-11ea-9cf2-81c9ad9dbb5d.png" alt="overview" width="600"/><br/>
+
+- `Solution` is the layer contains logic for how to select charge stations, such as
+  * whether need charge or not
+  * is destination reachable by single charge
+  * multiple charge station solution finding, which could be search along route or charge station based routing, implemented by lower layer
```

Comment:
lower layer -> internal layer might be better.

Reply:
changed.